Framework/CurveFitting/test/Algorithms/FitTest.h

```diff
@@ -789,6 +791,257 @@ class FitTest : public CxxTest::TestSuite {
+
+    TS_ASSERT_DELTA(outF->getParameter("B20"), 0.366336, 0.0001);
+    TS_ASSERT_DELTA(outF->getParameter("B22"), 3.98132, 0.0001);
+    TS_ASSERT_DELTA(outF->getParameter("B40"), -0.0304001, 0.0001);
+    TS_ASSERT_DELTA(outF->getParameter("B42"), -0.119605, 0.0001);
+    TS_ASSERT_DELTA(outF->getParameter("B44"), -0.130124, 0.0001);
+
+    ITableWorkspace_sptr output =
+        AnalysisDataService::Instance().retrieveWS<ITableWorkspace>(
+            "out_Workspace");
+    TS_ASSERT(output);
+    if (output) {
+      TS_ASSERT_EQUALS(output->rowCount(), 3);
+      TS_ASSERT_EQUALS(output->columnCount(), 4);
+      auto column = output->getColumn("Energy");
+      TS_ASSERT_DELTA(column->toDouble(0), 0.0, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(1), 29.3261, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(2), 44.3412, 0.0001);
+      column = output->getColumn("Intensity");
+      TS_ASSERT_DELTA(column->toDouble(0), 2.74937, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(1), 0.7204, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(2), 0.429809, 0.0001);
+      column = output->getColumn("Energy_calc");
+      TS_ASSERT_DELTA(column->toDouble(0), 0.0, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(1), 29.3261, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(2), 44.3412, 0.0001);
+      column = output->getColumn("Intensity_calc");
+      TS_ASSERT_DELTA(column->toDouble(0), 2.74937, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(1), 0.7204, 0.0001);
+      TS_ASSERT_DELTA(column->toDouble(2), 0.429809, 0.0001);
+    }
+  }
+
+  void getExpDecayMockData(Mantid::MantidVec &y, Mantid::MantidVec &e) {
+
+    y[0] = 5;
+    y[1] = 3.582656552869;
+    y[2] = 2.567085595163;
+    y[3] = 1.839397205857;
+    y[4] = 1.317985690579;
+    y[5] = 0.9443780141878;
+    y[6] = 0.6766764161831;
+    y[7] = 0.484859839322;
+    y[8] = 0.347417256114;
+    y[9] = 0.2489353418393;
+    y[10] = 0.1783699667363;
+    y[11] = 0.1278076660325;
+    y[12] = 0.09157819444367;
+    y[13] = 0.0656186436847;
+    y[14] = 0.04701781275748;
+    y[15] = 0.03368973499543;
+    y[16] = 0.02413974996916;
+    y[17] = 0.01729688668232;
+    y[18] = 0.01239376088333;
+
+    e.assign(19, 1.0);
+  }
+
+  void test_function_exp_decay_fit() {
+    Algorithms::Fit alg2;
+    TS_ASSERT_THROWS_NOTHING(alg2.initialize());
+    TS_ASSERT(alg2.isInitialized());
+
+    // create mock data to test against
+    std::string wsName = "ExpDecayMockData";
+    int histogramNumber = 1;
+    int timechannels = 20;
+    Workspace_sptr ws = WorkspaceFactory::Instance().create(
+        "Workspace2D", histogramNumber, timechannels, timechannels);
+    Workspace2D_sptr ws2D = boost::dynamic_pointer_cast<Workspace2D>(ws);
+    for (int i = 0; i < 20; i++)
+      ws2D->dataX(0)[i] = i;
+    Mantid::MantidVec &y = ws2D->dataY(0); // y-values (counts)
+    Mantid::MantidVec &e = ws2D->dataE(0); // error values of counts
+    getExpDecayMockData(y, e);
```

Comment:
The reason why the test was failing if you replaced

```
y[0] = 5;
y[1] = 3.582656552869;
...
```

by an initializer list is because you need an extra zero at the end, i.e., you can use an initializer list like this:

```
y = {5,
         3.582656552869,
         2.567085595163,
         1.839397205857,
         1.317985690579,
         0.9443780141878,
         0.6766764161831,
         0.484859839322,
         0.347417256114,
         0.2489353418393,
         0.1783699667363,
         0.1278076660325,
         0.09157819444367,
         0.0656186436847,
         0.04701781275748,
         0.03368973499543,
         0.02413974996916,
         0.01729688668232,
         0.01239376088333, 0};
```
and the test passes. The reason why you need an extra zero is because you are creating a workspace with `20` Y values (`timechannels=20`) but you are only initializing `19` elements (see method `getExpDecayMockData()`), adding this extra zero fixes the test. I've checked that you don't even need to care about errors, those can be left uninitialized and test will still pass. In addition, we don't need a separate method `getExpDecayMockData()`, y values can be initialized within the test.